packages/katello/nodejs-downshift/nodejs-downshift.spec

```
@@ -16,14 +16,14 @@ BuildRequires: %{?scl_prefix_nodejs}npm
 %else
 BuildRequires: nodejs-packaging
 %endif
-Requires: npm(@babel/runtime) >= 7.10.2
-Requires: npm(@babel/runtime) < 8.0.0
-Requires: npm(compute-scroll-into-view) >= 1.0.14
-Requires: npm(compute-scroll-into-view) < 2.0.0
-Requires: npm(prop-types) >= 15.7.2
-Requires: npm(prop-types) < 16.0.0
-Requires: npm(react-is) >= 16.13.1
-Requires: npm(react-is) < 17.0.0
+Requires: %{?scl_prefix}npm(babel-runtime) >= 7.10.2
+Requires: %{?scl_prefix}npm(babel-runtime) < 8.0.0
```

This part was a wrong suggestion.
```suggestion
Requires: %{?scl_prefix}npm(babel/runtime) >= 7.10.2
Requires: %{?scl_prefix}npm(babel/runtime) < 8.0.0
```

---

are you sure? https://github.com/theforeman/foreman-packaging/blob/rpm/develop/packages/foreman/nodejs-babel-runtime/nodejs-babel-runtime.spec#L4

---

That's not how it's supposed to work and a bug in the babel runtime packaging ...

---

in any case it doesn't seem to fix the issue. `nodejs-symlink-deps` looks at the `package.json`, not the spec file.

---

Yes `Requires` are unrelated in the build process (unlike `BuildRequires`). They are only added after the fact and are used by other packages.

---

where would it get `@babel/runtime` installed from though? It's not part of the tarball, could it be installed by the npm `BuildRequires`? 
also, it seems to be linking to `/opt/theforeman/tfm/root/usr/lib/node_modules` and not `${BUILD_ROOT}/opt/theforeman/tfm/root/usr/lib/node_modules`, not sure if that's expected? could it be some leftovers from some other build?

---

Yes, that may be an issue. The NodeJS packaging is painful. The NPM packages are quite often **very** bad themselves and then add to that that the RPM layer on top of it is also mediocre. Like this example, we've found that it appears we're the only ones who actually create NPM packages in SCL.